pkg/reporter/cyclonedx.go

```diff
@@ -0,0 +1,240 @@
+package reporter
+
+import (
+	"fmt"
+	"os"
+	"sync"
+	"time"
+
+	cdx "github.com/CycloneDX/cyclonedx-go"
+	"github.com/google/uuid"
+	"github.com/safedep/dry/utils"
+	"github.com/safedep/vet/pkg/analyzer"
+	"github.com/safedep/vet/pkg/common/logger"
+	commonUtils "github.com/safedep/vet/pkg/common/utils"
+	"github.com/safedep/vet/pkg/common/utils/regex"
+	"github.com/safedep/vet/pkg/models"
+	"github.com/safedep/vet/pkg/policy"
+	"github.com/safedep/vet/pkg/readers"
+	"github.com/safedep/vet/pkg/reporter/data"
+)
+
+type CycloneDXToolMetadata struct {
+	Name    string
+	Version string
+	Purl    string
+}
+
+// CycloneDXReporterConfig contains configuration parameters for the CycloneDX reporter
+type CycloneDXReporterConfig struct {
+	Tool CycloneDXToolMetadata
+
+	// Path defines the output file path
+	Path string
+
+	// Application component name, this is the top-level component in the BOM
+	ApplicationComponentName string
+
+	// Unique identifier for this BOM confirming to UUID RFC 4122 standard
+	// If empty, a new UUID will be generated
+	SerialNumber string
+}
+
+type cycloneDXReporter struct {
+	sync.Mutex
+	config CycloneDXReporterConfig
+	bom    *cdx.BOM
+}
+
+var cdxUUIDRegexp = regex.MustCompileAndCache(`^urn:uuid:[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$`)
```

Comment:
Does the CDX spec says that the UUID need to be in this format?

Reply:
yes, i've taken regex from their spec - https://cyclonedx.org/docs/1.6/json/#serialNumber